parsec/core/cli/freeze_user.py

```diff
@@ -0,0 +1,133 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+from __future__ import annotations
+
+from typing import TypedDict, cast
+
+import click
+
+from parsec.api.protocol import OrganizationID, UserID
+from parsec.api.rest import (
+    freeze_user_rep_serializer,
+    freeze_user_req_serializer,
+    user_list_rep_serializer,
+)
+from parsec.cli_utils import cli_exception_handler
+from parsec.core.backend_connection.transport import http_request
+from parsec.core.cli.utils import cli_command_base_options
+from parsec.core.types import BackendAddr
+from parsec.utils import trio_run
+
+
+class User(TypedDict):
+    user_name: str
+    user_id: UserID
+    user_email: str
+    frozen: bool
+
+
+def show_user(user: User, prefix: str = "") -> None:
+    user_name = click.style(user["user_name"], fg="yellow")
+    user_id = click.style(user["user_id"].str, fg="yellow")
+    user_email = click.style(user["user_email"], fg="yellow")
+    frozen = user["frozen"]
+    frozen = click.style("Frozen", fg="red") if frozen else click.style("Not frozen", fg="yellow")
+    empty_prefix = " " * len(prefix)
+    click.echo(f"{prefix}{user_name} <{user_email}> ")
+    click.echo(f"{empty_prefix}- Parsec ID: {user_id} ")
+    click.echo(f"{empty_prefix}- Status: {frozen} ")
+
+
+async def _list_users(
+    organization_id: OrganizationID, backend_addr: BackendAddr, administration_token: str
+) -> None:
+    url = backend_addr.to_http_domain_url(
+        f"/administration/organizations/{organization_id.str}/users"
+    )
+
+    rep_data = await http_request(
+        url=url, method="GET", headers={"authorization": f"Bearer {administration_token}"}
+    )
+
+    cooked_rep_data = user_list_rep_serializer.loads(rep_data)
+    users = cooked_rep_data["users"]
+    for user in users:
+        show_user(cast(User, user), prefix="• ")
+        click.echo("")
+    number_of_users = click.style(len(users), fg="yellow")
+    click.echo(f"Total users: {number_of_users}")
+
+
+@click.command(short_help="List users of an organization")
+@click.argument("organization_id", required=True, type=OrganizationID)
+@click.option("--addr", "-B", required=True, type=BackendAddr.from_url, envvar="PARSEC_ADDR")
+@click.option("--administration-token", "-T", required=True, envvar="PARSEC_ADMINISTRATION_TOKEN")
+@cli_command_base_options
+def list_users(
+    organization_id: OrganizationID,
+    addr: BackendAddr,
+    administration_token: str,
+    debug: bool,
+    **kwargs: object,
+) -> None:
+    """List users of an organization using the administration API.
+
+    The following information is returned for each user:
+
+    \b
+    • USER_NAME <USER_EMAIL>
+      - Parsec ID: USER_ID
+      - Status: Frozen/Not frozen
```

Comment:
I wonder if this might not be annoying in case you want to `grep` the output :thinking: (typically for large organizations)

Maybe we can add an option (similar to `git log --oneline` or `git rev-parse --short`) to have a one-line-per-user output:
```
$ parsec core list_users --oneline -B "parsec://$server" -T $administration_token $organization
Alice <alice@example.com> (67ee640058aa45ca9281717c866baa06) Not frozen
Bob <bob@example.com> (0d22530361484b86b28e5b2c3a089772) Frozen
David <david@example.com> (1f1b81052fa947babdbacd6b147b3622) Not Frozen

$ parsec core list_users --oneline -B "parsec://$server" -T $administration_token $organization | grep "Frozen"
Bob <bob@example.com> (0d22530361484b86b28e5b2c3a089772) Frozen
```

IDK what would be the better format though:
- maybe Parsec ID could be first column, because same length? 
- maybe we can use `\t` for separator?
- maybe there is a `click` option to format output as a table?

Reply:
I'll answer myself: actually we can `grep` lines before (`-B`) and after (`-A`). So something like `grep -i -B 2 frozen` should be enough to filter by frozen status.